zipline/_protocol.pyx

```diff
@@ -29,7 +39,8 @@ cdef class BarData:
 
     This is what is passed as `data` to the `handle_data` function.
     """
-    def __init__(self, data_portal, simulation_dt_func, data_frequency):
+    def __init__(self, data_portal, simulation_dt_func, data_frequency,
+                 universe_func=None):
```

Comment:
by a bunch of tests.  I prefer not requiring a universe func because in the future, we won't pass one when the old API is deprecated.